libtransmission/rpcimpl.c

```diff
@@ -927,6 +940,26 @@ static char const* torrentGet(tr_session* session, tr_variant* args_in, tr_varia
 ****
 ***/
 
+static char const* setLabels(tr_torrent* tor, tr_variant* list)
+{
+    int i;
+    const char * str;
+    size_t str_len;
+    const int n = tr_variantListSize(list);
+    tr_ptrArrayForeach(&tor->labels, tr_free);
+    tr_ptrArrayClear(&tor->labels);
+    for (i = 0; i < n; i++)
+    {
+        if (tr_variantGetStr(tr_variantListChild(list, i), &str, &str_len) && str && str_len)
+        {
+            if (strchr(str, ',') != NULL)
+                return "labels can not contain comma (,) character";
```

Comment:
* _(nit)_ 'cannot' should be one word

* this verify-and-return-error-message-on-failure should happen _before_ we clear out the previous labels